### ISSUE TITLE
Enable ed25519 support in CLI for Public Preview

### ISSUE THREAD

**Issue by kamusta-msft** [Azure/azure-cli#27708]
### Preconditions

- [X] No need to upgrade Python SDK or the Python SDK is ready.

### Related command

https://learn.microsoft.com/en-us/cli/azure/vm?view=azure-cli-latest#az-vm-create (--generate-ssh-keys).

### Resource Provider

Microsoft.Compte

### Description of Feature or Work Requested

We're modifying the generateSshKeyPair API and need to update the CLI to reflect those changes. The generateSshKeyPair now has a response body that takes in the "encryptionType" property. The allowed values of this property are either "RSA" or "Ed25519". Looking at our existing docs, this cli will need to get updated: https://learn.microsoft.com/en-us/cli/azure/vm?view=azure-cli-latest#az-vm-create (--generate-ssh-keys). One possible update would be to turn this from a true/false to a flag that requires a property. I've pasted the Swagger Spec below:

### Minimum API Version Required

2023-09-01

### Swagger PR link / SDK link

https://github.com/Azure/azure-rest-api-specs/pull/26090

SDK will get generated by Swagger team after all changes are merged

### Request Example

https://github.com/Azure/azure-rest-api-specs/blob/16f37d09a00385dd4143cbc2a07b0e860e1e018e/specification/compute/resource-manager/Microsoft.Compute/ComputeRP/stable/2023-09-01/examples/sshPublicKeyExamples/SshPublicKey_GenerateKeyPair_EncryptionWithRSA.json

Sending "encryptionType" in the request body for the generateKey API

### Target Date

2023-12-30

### PM Contact

@srijangupta

### Engineer Contact

@kamusta

**Comment by yonzhan** [comment]
Thank you for opening this issue, we will look into it.

**Comment by kamusta-msft** [comment]
@srijang to confirm the target date

**Comment by yanzhudd** [comment]
Hi @kamusta-msft let me confirm with you if this feature request is actually the same feature as #27449

**Comment by yanzhudd** [comment]
Hi @kamusta-msft,
CLI is developed base on [Azure SDK](https://github.com/Azure/azure-sdk-for-python), but it seems that there isn't api-version 2023-09-01 in the latest version of [compute module](https://github.com/Azure/azure-sdk-for-python/tree/main/sdk/compute/azure-mgmt-compute/azure/mgmt/compute). 
Please refer to the doc of [How to Request a Feature in SDK](https://github.com/Azure/azure-sdk-for-python/blob/main/doc/dev/how_to_request_a_feature_in_sdk.md) to release a new version of SDK with api-version 2023-09-01 firstly, only after that can we develop this new feature. Thanks.

**Comment by kamusta-msft** [comment]
@yanzhudd Confirmed, same as https://github.com/Azure/azure-cli/issues/27449

The API specs for 2023-09-01 are currently in the process of getting merged into master by the Swagger team. Once merged, they'll generate the SDK. We created this feature request so that it can be considered for feature/sprint planning

**Comment by yanzhudd** [comment]
> @yanzhudd Confirmed, same as #27449
> 
> The API specs for 2023-09-01 are currently in the process of getting merged into master by the Swagger team. Once merged, they'll generate the SDK. We created this feature request so that it can be considered for feature/sprint planning

@kamusta-msft, okay, we'll develop this feature once the SDK is released.

**Comment by yanzhudd** [comment]
> One possible update would be to turn this from a true/false to a flag that requires a property.

Hi @kamusta-msft, 
in order to avoid a breaking change, it is suggested to add a new parameter (e.g., `--ssh-ecryption-type`) to specify the encryption type of the SSH keys to be generated. Could you please confirm if it works for you?

**Comment by yanzhudd** [comment]
Hi @kamusta-msft,

could you please help confirm these questions:
1. In fact, CLI command `az vm create --generate-ssh-keys` would not call the generateSshKeyPair API, instead using paramiko package to generate keys. It is the command `az sshkey create` which call the generateSshKeyPair API. Thus, does the following change meets your expectation?
a. CLI modifies the command `az sshkey create -n keyName` to support Ed25519 generation
b. CLI would not change the parameters of `az vm create`
Users can use `az sshkey create -n keyName --encryption-type Ed25519` to generate Ed25519 key, and then use `az vm create --ssh-key-name keyName` to create vm with generated Ed25519 key. 

2. When we call generateSshKeyPair API with Ed25519 SSH Key, service side reported an error as:
![image](https://github.com/Azure/azure-cli/assets/105691024/fda57cb2-e128-41df-a96a-9ce8bb30006a)
Could you please help register our subId '0b1f6471-1bf0-4dda-aec3-cb9272f09590', or is there any way we can register by ourselves?

Thanks!

**Comment by kamusta-msft** [comment]
> az vm create --generate-ssh-keys would not call the generateSshKeyPair API, instead using paramiko package to generate keys
which encryption format are the keys in? If it's only RSA, then we may need to update the documentation. 

The other points you mentioned seem fine to me. @srijang to give the final confirmation

2. Your sub should be registered now.
